tests/seeding/test_find_or_create.py

```diff
@@ -127,6 +130,42 @@ def list(self, page: Optional[int] = None, per_page: int = 100):
         datasets.register(Dataset(duplicate_name, "dup", "duplicate"))
     return datasets
 
+@pytest.fixture
+def predictor_collection() -> PredictorCollection:
+    class SeedingTestPredictorCollection(PredictorCollection):
+        predictors = []
+
+        def register(self, model: SimpleMLPredictor) -> SimpleMLPredictor:
+            self.predictors.append(model)
+            return model
+        
+        def update(self, model):
+            self.predictors = [r for r in self.predictors if r.uid != model.uid]
+            return self.register(model)
+
+        def list(self, page: Optional[int] = None, per_page: int = 100):
+            if page is None:
+                return self.predictors
+            else:
+                return self.predictors[(page - 1)*per_page:page*per_page]
+
+    predictors = SeedingTestPredictorCollection(UUID('6b608f78-e341-422c-8076-35adc8828545'),
+                                session)
+
+    for i in range(0, 5):
```

Comment:
The pre-registration logic here could use a comment -- its no obvious what its trying to accomplish or how the test will use it.

Reply:
@maxhutch Thanks for the speedy review and specific pointers! 
the latest commits should address all your requests.